File: frontend/src/api/docs.ts
import { apiClient } from './client'

export type StaticDocsSearchResult = {
  title: string
  source_path: string
  section_heading: string | null
  snippet: string
  matched_terms: string[]
  line_start: number
  line_end: number
}

export type StaticDocsSearchResponse = {
  query: string
  count: number
  results: StaticDocsSearchResult[]
  limitations: string[]
}

export async function searchDocs(query: string, maxResults = 8) {
  const response = await apiClient.get<StaticDocsSearchResponse>(
    '/api/v1/docs/search',
    {
      params: {
        q: query,
        max_results: maxResults,
      },
    },
  )

  return response.data
}

File: backend/app/schemas/docs.py
from pydantic import BaseModel


class StaticDocsSearchResult(BaseModel):
    title: str
    source_path: str
    section_heading: str | None = None
    snippet: str
    matched_terms: list[str]
    line_start: int
    line_end: int


class StaticDocsSearchResponse(BaseModel):
    query: str
    count: int
    results: list[StaticDocsSearchResult]
    limitations: list[str]


class StaticDocsChunkPreview(BaseModel):
    chunk_id: str
    source_path: str
    title: str
    section_heading: str | None = None
    text: str
    line_start: int
    line_end: int
    character_count: int
    content_hash: str


class StaticDocsChunkPreviewResponse(BaseModel):
    count: int
    chunks: list[StaticDocsChunkPreview]
    limitations: list[str]
